src/bootstrap.php

```diff
@@ -0,0 +1,293 @@
+};
+
+$app['negotiator'] = function () {
+    return new VersionedNegotiator();
+};
+
+$app->get('/recommendations/{type}/{id}', function (Request $request, string $type, string $id) use ($app) {
+    try {
+        $identifier = Identifier::fromString("{$type}/{$id}");
+
+        if ('article' !== $type) {
+            throw new BadRequestHttpException('Not an article');
+        }
+    } catch (InvalidArgumentException $e) {
+        throw new NotFoundHttpException();
+    }
+
+    $accepts = [
+        'application/vnd.elife.recommendations+json; version=1',
+    ];
+
+    /** @var Accept $type */
+    $type = $app['negotiator']->getBest($request->headers->get('Accept'), $accepts);
+
+    $version = (int) $type->getParameter('version');
+    $type = $type->getType();
+
+    $page = $request->query->get('page', 1);
+    $perPage = $request->query->get('per-page', 20);
+
+    $article = $app['elife.api_sdk']->articles()->getHistory($id);
+
+    // Do this early as we have to then request each episode.
+    $podcastEpisodes = $app['elife.api_sdk']->podcastEpisodes()
+        ->containing(Identifier::article($id))
+        ->slice(0, 100);
+
+    try {
+        $article = $article->wait()->getVersions()[0];
+    } catch (BadResponse $e) {
+        switch ($e->getResponse()->getStatusCode()) {
+            case Response::HTTP_GONE:
+            case Response::HTTP_NOT_FOUND:
+                throw new HttpException($e->getResponse()->getStatusCode(), "$identifier does not exist", $e);
+        }
+
+        throw $e;
+    }
+
+    $relations = $app['elife.api_sdk']->articles()
+        ->getRelatedArticles($id)
+        ->sort(function (Article $a, Article $b) {
+            static $order = [
+                'retraction' => 1,
+                'correction' => 2,
+                'external-article' => 3,
+                'registered-report' => 4,
+                'replication-study' => 5,
+                'research-advance' => 6,
+                'scientific-correspondence' => 7,
+                'research-article' => 8,
+                'tools-resources' => 9,
+                'feature' => 10,
+                'insight' => 11,
+                'editorial' => 12,
+                'short-report' => 13,
+            ];
+
+            return $order[$a->getType()] <=> $order[$b->getType()];
+        });
+
+    $collections = $app['elife.api_sdk']->collections()
+        ->containing($article->getIdentifier())
+        ->slice(0, 100);
+
+    $podcastEpisodeChapters = $podcastEpisodes
+        ->reduce(function (Sequence $chapters, PodcastEpisode $episode) use ($article) {
+            foreach ($episode->getChapters() as $chapter) {
+                foreach ($chapter->getContent() as $content) {
+                    if ($article->getId() === $content->getId()) {
+                        $chapters = $chapters->append(new PodcastEpisodeChapterModel($episode, $chapter));
+                        continue 2;
```

Comment:
It continues the outer `foreach`, so once it finds the article in a chapter it goes to the next chapter.